### PR TITLE
Upgrade lit deps

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Closes https://github.com/withastro/astro/issues/7230

## Testing

N/A

## Docs

N/A

### CODE REVIEW

File: packages/integrations/lit/hydration-support.js [hunk 1]
```diff
@@ -1 +1,2 @@
-import 'lit/experimental-hydrate-support.js';
+// @ts-check
+import '@lit-labs/ssr-client/lit-element-hydrate-support.js';
```
Comment (bluwy) [inline packages/integrations/lit/hydration-support.js:2]: Do we need to add `@lit-labs/ssr-client` into `dependencies`?

File: packages/integrations/lit/package.json [hunk 2]
```diff
@@ -42,21 +42,21 @@
...
     "cheerio": "^1.0.0-rc.11",
-    "lit": "^2.7.0",
+    "lit": "^2.7.5",
     "mocha": "^9.2.2",
     "sass": "^1.52.2"
   },
   "peerDependencies": {
     "@webcomponents/template-shadowroot": "^0.2.1",
-    "lit": "^2.7.0"
+    "lit": "^2.7.5"
```
Comment (bluwy) [inline packages/integrations/lit/package.json:60]: Updating the peer deps would be a breaking change (and a major). Do we need to bump the `lit` version here?

File: pnpm-lock.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (bluwy) [inline pnpm-lock.yaml:1]: I think you'd need to use pnpm 8.6 and above to generate the lockfile v6.1

File: packages/integrations/lit/package.json (hunk 2 above)
Comment (matthewp) [reply packages/integrations/lit/package.json:60]: No, we don't nee to update this one, will downgrade.

File: pnpm-lock.yaml (hunk 3 above)
Comment (matthewp) [reply pnpm-lock.yaml:1]: Hm, very odd:

> pnpm -v
> 8.6.2

Comment (matthewp) [reply pnpm-lock.yaml]: Got it.

**Review by ematipico — APPROVED** [review]
Looks good to me, after addressing @bluwy 's comment

**Pull request merged**